config/proxy.go

```diff
@@ -78,3 +80,14 @@ func (c *Config) setProxy() error {
 
 	return nil
 }
+
+func parseBucketLookupType(typeStr string) minio.BucketLookupType {
+	valMap := map[string]minio.BucketLookupType{
+		"auto": minio.BucketLookupAuto,
+		"dns":  minio.BucketLookupDNS,
+		"path": minio.BucketLookupPath,
+	}
+
+	// also when not found the type, return "auto" type.
+	return valMap[typeStr]
```

Comment:
I think this function should probably check if `typeStr` is in the map, and return an error if it isn't. And this error checking should be done in `validateConfig()` in config/config.go

Reply:
I think we only need to check it in `validateConfig()`.

Reply:
Also I refactored it again.